dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java

```diff
@@ -681,6 +682,7 @@ public Result deleteProcessDefinitionByCode(@ApiIgnore @RequestAttribute(value =
     @ResponseStatus(HttpStatus.OK)
     @ApiException(BATCH_DELETE_PROCESS_DEFINE_BY_CODES_ERROR)
     @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
+    @Transactional
```

Comment:
I think we should not add this annotation in controller layer.
It would better to create a batch delete method in service if we want to do that.

Reply:
BTW, I don't think this batch delete action should be transactional, because different workflows are independent of each other, not an atomic relationship.
WDYT? @zhongjiajie @ruanwenjun @SbloodyS

Reply:
> BTW, I don't think this batch delete action should be transactional, because different workflows are independent of each other, not an atomic relationship. WDYT? @zhongjiajie @ruanwenjun @SbloodyS

+1